model/Hierarchy.php

```diff
@@ -602,6 +615,17 @@ public function numChildren($cache = true) {
 	}
 
 	/**
+	 * Checks if we're on a controller where we should filter. ie. Are we loading the SiteTree?
+	 *
+	 * @return bool
+	 */
+	public function showingCMSTree() {
+		$controller = Controller::curr();
```

Comment:
`Controller::curr()` throws an error if none is assigned. You can add a `Controller::has_curr()` check to protect against that. :)